### PR TITLE
Revamp the dynamodb connector

### PR DESCRIPTION
**Pull request by Bhashinee**
## Purpose
Revamp the dynamodb connector.

This will fix : https://github.com/ballerina-platform/ballerina-library/issues/4831, https://github.com/ballerina-platform/ballerina-library/issues/4830, https://github.com/ballerina-platform/ballerina-library/issues/5800

### CODE REVIEW

File: ballerina/client.bal [hunk 1]
```diff
@@ -46,81 +46,91 @@ public isolated client class Client {
...
 
-    # Create a table. The CreateTable operation adds a new table to your account. In an AWS account, table names must be
+    # Creates a table. The CreateTable operation adds a new table to your account. In an AWS account, table names must be
     # unique within each Region. That is, you can have two tables with same name if you create the tables in diﬀerent
     # Regions.
     #
     # + tableCreationRequest - The request payload to create a table
-    # + return - If success, dynamodb:TableCreateResponse record, else an error
-    remote isolated function createTable(TableCreateRequest tableCreationRequest) returns TableCreateResponse|error {
+    # + return - If success, dynamodb:CreateTableOutput record, else an error
```
Comment (daneshk) [inline ballerina/client.bal:57]:
```suggestion
    # + return - If success, dynamodb:TableDescription record, else an error
```

File: ballerina/client.bal [hunk 2]
```diff
@@ -46,81 +46,91 @@ public isolated client class Client {
...
+        return tableDescription.cloneWithType(TableDescription);
     }
 
-    # Delete a table.
+    # Deletes a table.
     #
     # + tableName - The name of the table to delete
-    # + return - If success, dynamodb:TableDeleteResponse record, else an error
-    remote isolated function deleteTable(string tableName) returns TableDeleteResponse|error {
+    # + return - If success, dynamodb:DeleteTableOutput record, else an error
```
Comment (daneshk) [inline ballerina/client.bal:74]:
```suggestion
    # + return - If success, dynamodb:TableDescription record, else an error
```

File: ballerina/client.bal [hunk 3]
```diff
@@ -46,81 +46,91 @@ public isolated client class Client {
...
+        return tableDescription.cloneWithType(TableDescription);
     }
 
-    # Describe a table.
+    # Describes a table.
     #
     # + tableName - The name of the table to delete
-    # + return - If success, dynamodb:TableDescribeResponse record, else an error
-    remote isolated function describeTable(string tableName) returns TableDescribeResponse|error {
+    # + return - If success, dynamodb:DescribeTableOutput record, else an error
```
Comment (daneshk) [inline ballerina/client.bal:93]: same here

File: ballerina/records.bal [hunk 4]
```diff
@@ -41,845 +41,752 @@ public type AwsCredentials record {
...
+    int? accountMaxWriteCapacityUnits?;
+    # The maximum read capacity units that your account allows you to provision for a new table that you are creating in this Region, including the read capacity units
+    int? tableMaxReadCapacityUnits?;
+    # The maximum write capacity units that your account allows you to provision for a new table that you are creating in this Region, including the write capacity units
+    # provisioned for its global secondary  indexes (GSIs).
+    int? tableMaxWriteCapacityUnits?;
+};
+
+# Represents the response after `WriteBatchItem` operation.
+public type WriteBatchItemOutput record {
```
Comment (daneshk) [inline ballerina/records.bal:76]: Shall we rename the record, How about `BatchItemPostOutput`. is this for creating a new item or updating an item?

Comment (daneshk) [reply ballerina/records.bal:76]: just trying to avoid starting with a verb. is this name got from dynamodb API?

File: ballerina/records.bal [hunk 5]
```diff
@@ -41,845 +41,752 @@ public type AwsCredentials record {
...
-#
-# + ConsumedCapacity - The capacity units consumed by the Scan operation.
-# + Count - The number of items in the response.
-# + Items - An array of item attributes that match the scan criteria. Each element in this array consists of an
-#           attribute name and the value for that attribute.
-# + LastEvaluatedKey - The primary key of the item where the operation stopped, inclusive of the previous result set.
-#                      Use this value to start a new operation, excluding this value in the new request.
-# + ScannedCount - The number of items evaluated, before any ScanFilter or QueryFilter is applied. A high ScannedCount 
-#                  value with few, or no, Count results indicates an ineﬃcient Scan or Query operation.
 public type QueryOrScanResponse record {
```
Comment (daneshk) [inline ballerina/records.bal:194]: For some records, we ended up with `Input`/`Output`, and for some records we used `Request`/`Response`. Shall we make it consistent?

File: docs/spec/spec.md [hunk 6]
```diff
@@ -0,0 +1,288 @@
...
+        12. [getBatchItem](#getBatchItem)
+        13. [writeBatchItem](#writeBatchItem)
+        14. [describeLimits](#describeLimits)
+        15. [createBackup](#createBackup)
+        16. [deleteBackup](#deleteBackup)
+        17. [getTTL](#getTTL)
+       
+## 1. [Overview](#1-overview)
+
+The Ballerina `dynamodb` library facilitates APIs to allow you to access the Amazon DynamoDB REST API. Amazon DynamoDB is a fully managed NoSQL database service that provides fast and predictable performance with seamless scalability. AmazonDynamoDB enables customers to offload the administrative burdens of operating and scaling distributed databases to AWS, so they do not have to worry about hardware provisioning, setup and configuration, replication, software patching, or cluster scaling.
```
Comment (daneshk) [inline docs/spec/spec.md:45]:
```suggestion
The Ballerina `dynamodb` library facilitates APIs to allow you to access the Amazon DynamoDB REST API. Amazon DynamoDB is a fully managed NoSQL database service that provides fast and predictable performance with seamless scalability. Amazon DynamoDB enables customers to offload the administrative burdens of operating and scaling distributed databases to AWS, so they do not have to worry about hardware provisioning, setup, and configuration, replication, software patching, or cluster scaling.
```

File: docs/spec/spec.md [hunk 7]
```diff
@@ -0,0 +1,288 @@
...
+
+This API can be used to create a new table in DynamoDB with the specified parameters.
+
+```ballerina
+# Creates table in your account. The CreateTable operation adds a new table to your account. In # an AWS account, table names must be unique within each Region. That is, you can have two # tables with same name if you create the tables in diﬀerent Regions.
+#
+#
+# + tableCreationRequest - The request payload to create a table
+# + return - If success, dynamodb:CreateTableOutput record, else an error
+remote isolated function createTable(CreateTableInput tableCreationRequest) returns TableDescription|error {
```
Comment (daneshk) [inline docs/spec/spec.md:91]:
```suggestion
remote isolated function createTable(CreateTableInput tableCreationInput) returns TableDescription|error {
```

File: docs/spec/spec.md [hunk 8]
```diff
@@ -0,0 +1,288 @@
...
+
+#### [describeTable](#describeTable)
+
+This API can be used to get the table details in DynamoDB with the given table name.
+
+```ballerina
+# Describes a table.
+#
+# + tableName - The name of the table to delete
+# + return - If success, dynamodb:DescribeTableOutput record, else an error
```
Comment (daneshk) [inline docs/spec/spec.md:114]:
```suggestion
# + return - If success, dynamodb:TableDescription record, else an error
```

File: docs/spec/spec.md [hunk 9]
```diff
@@ -0,0 +1,288 @@
...
+#### [updateTable](#updateTable)
+
+This API can be used to update the entries in the table with the specified parameters.
+
+```ballerina
+# Updates a table.
+#
+# + tableUpdateRequest - The request payload to update a table 
+# + return - If success, dynamodb:UpdateTableOutput record, else an error
+remote isolated function updateTable(UpdateTableInput tableUpdateRequest) returns TableDescription|error {
```
Comment (daneshk) [inline docs/spec/spec.md:138]:
```suggestion
# + return - If success, dynamodb:TableDescription record, else an error
remote isolated function updateTable(UpdateTableInput tableUpdateInput) returns TableDescription|error {
```

File: docs/spec/spec.md [hunk 10]
```diff
@@ -0,0 +1,288 @@
...
+
+```ballerina
+# Creates a new item, or replaces an old item with a new item. If an item that has the same primary key as the new
+# item already exists in the speciﬁed table, the new item completely replaces the existing item. You can perform a
+# conditional put operation (add a new item if one with the speciﬁed primary key doesn't exist), or replace an
+# existing item if it has certain attribute values.
+#
+# + request - The request payload to create an item
+# + return - If success, dynamodb:ItemDescription record, else an error
+remote isolated function createItem(CreateItemInput request) returns ItemDescription|error {
```
Comment (daneshk) [inline docs/spec/spec.md:153]:
```suggestion
remote isolated function createItem(ItemCreateInput request) returns ItemDescription|error {
```
Shall we do like this `<subject><verb><input/output>` for record definitions? WDYT?

Comment (Bhashinee) [reply docs/spec/spec.md]: All the comments are fixed in [15cc65b](https://github.com/ballerina-platform/module-ballerinax-aws.dynamodb/pull/31/commits/15cc65b697e9cf7c20776e57d182b98f6cba6684)

File: examples/game-scores/client.bal [hunk 11]
```diff
@@ -0,0 +1,82 @@
+import ballerina/io;
```
Comment (daneshk) [inline examples/game-scores/client.bal:1]: license headers?

**Review by daneshk — DISMISSED** [review]
LGTM

**Review by daneshk — APPROVED** [review]

**Pull request merged**